include/zephyr/bluetooth/ead.h

```diff
@@ -0,0 +1,147 @@
+ *
+ * @retval size_t Total size in bytes of the advertising structure from the
+ *                `bt_data` structure set.
+ */
+size_t bt_ead_get_encrypted_data_size(const struct bt_data *ad_structs, size_t num_elts);
+
+/**
+ * @brief Helper function to get the size (in bytes) of the advertising
+ * structure inside the AD Data of an encrypted advertising structure.
+ *
+ * @param[in] payload Advertising structure of type `BT_DATA_ENCRYPTED_AD_DATA`.
+ *
+ * @retval size_t Size in bytes of the advertising structure inside the AD data
+ *                of the @p payload advertising structure. If @p payload is not
+ *                an encrypted advertising structure, the function return 0.
+ */
+size_t bt_ead_get_decrypted_data_size(const struct bt_data *payload);
+
+/**
+ * @brief Helper function to convert `bt_data` structure into an advertising
+ * structure in an array. The data are formatted according to the Bluetooth Core
+ * Specification v. 5.4, vol. 3, part C, 11.
+ *
+ * @param[in]  input Single `bt_data` structure to read from.
+ * @param[out] output_ad_structure Buffer large enough to store the advertising
+ *             structure in @p input. The size of it must be at least the size
+ *             of the `AD data + 2` (for the type and the length).
+ *
+ * @retval size_t Number of bytes written in @p output_ad_structure.
+ */
+size_t bt_data_to_ad_structure(const struct bt_data *input, uint8_t *output_ad_structure);
+
+#ifdef CONFIG_BT_TESTING
+
+/**
+ * @brief This is the same function as @ref bt_ead_encrypt except this one add
+ * the @p randomizer parameter to let the user set the randomizer value.
+ *
+ * @note This function is only used for testing purpose and is only available
+ *       with `CONFIG_BT_TESTING` enabled.
+ *
+ * @param[in]  session_key Key of 16 bytes used for the encryption.
+ * @param[in]  iv Initialisation Vector used to generate the nonce. It must be
+ *             changed each time the Session Key change.
+ * @param[in]  randomizer Randomizer value used to generate the nonce. The value
+ *             is also placed in frond of the encrypted advertising data.
+ * @param[in]  payload Advertising Data to encrypt. Can be multiple advertising
+ *             structure concatenated.
+ * @param[in]  payload_size Size of the Advertising Data to encrypt.
+ * @param[out] encrypted_payload Encrypted Ad Data including the Randomizer and
+ *             the MIC. Size must be at least
+ *             `BT_EAD_RANDOMIZER+payload_size+BT_EAD_MIC_SIZE`.
+ *
+ * @retval 0 Data have been correctly encrypted and authenticated.
+ * @retval -EIO Error occurred during the encryption or the authentication.
+ * @retval -EINVAL One of the argument is a NULL pointer.
+ * @retval -ECANCELED Error occurred during the random number generation.
+ */
+int bt_test_ead_encrypt(const uint8_t *session_key, const uint8_t *iv, const uint8_t *randomizer,
+			const uint8_t *payload, size_t payload_size, uint8_t *encrypted_payload);
```

Comment:
I still suggest that for the fixed size arrays, we denote the size of the arrays in the function declaration

Reply:
Yes, I forgot to do it, but will do.